doc/performance-investigations/0023-upgrade-python-to-3.11.4/summary.md

```diff
@@ -0,0 +1,33 @@
+# Python 3.11.4 Upgrade investigation
+
+EQ Questionaire Runner is currently on `3.10.8` .
+This investigation is to see the performance of Runner when upgraded to `3.11.4` compared to `3.10.8` .
+
+https://github.com/ONSdigital/eq-questionnaire-runner/tree/upgrade-python-version-311
+
+## Benchmark profile
+
+| Option | Value |
+|--------|-------|
+| Requests file | requests/test_benchmark_business.json|
+| Run time | 20m |
+| User wait time minimum | 1s |
+| User wait time maximum | 2s |
+| Clients | 50 |
+| Hatch rate | 50 |
+
+## Results
+
+Results based on 99th percentile timings.
+| Metric              | Baseline | Investigation |
+| ------------------- | -------- | ------------- |
+| Questionnaire GETs  | 827ms    | 826ms         |
+| Questionnaire POSTs | 565ms    | 539ms         |
+| All requests        | 730ms    | 720ms         |
+| Total Requests      | 61,358   | 61,269        |
+| Total Failures      | 0 .      | 0             |
```

Comment:
Is this correct? I know not many but looks like there were some failures for the investigation?

Reply:
I had some gcp glitches which caused 0.3% errors, re-run a smaller benchmark after to confirm and it was error free but I will run another full scale one now and add the results here.

Reply:
New investigation benchmark (uninterrupted by gcp glitches) results added now.